sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/UrlTests.groovy

```diff
@@ -0,0 +1,98 @@
+package com.azure.storage.file.datalake
+
+class UrlTests extends APISpec {
+
+    def "test custom url"() {
+        when:
+        def customUrl = "https://www.customstorageurl.com"
+        DataLakeServiceClient client = new DataLakeServiceClientBuilder()
+            .endpoint(customUrl)
+            .credential(primaryCredential)
+            .buildClient()
+        then:
+        client.getAccountUrl() == client.blobServiceClient.getAccountUrl()
+    }
+
+    def "test short host url"() {
+        when:
+        def shortHostUrl = "https://account.core.windows.net"
```

Comment:
Is this really a valid azure url?

Reply:
not a valid Azure URL, just testing an edge case